aten/src/ATen/core/NamedRegistrations.cpp

```diff
@@ -0,0 +1,458 @@
+  m.impl("sum.DimnameList_out", CppFunction::makeFallthrough());
+  m.impl("sum.IntList_out", CppFunction::makeFallthrough());
+  m.impl("sum.dim_DimnameList", CppFunction::makeFallthrough());
+  m.impl("sum.dim_IntList", CppFunction::makeFallthrough());
+  m.impl("t", CppFunction::makeFallthrough());
+  m.impl("tan", CppFunction::makeFallthrough());
+  m.impl("tan.out", CppFunction::makeFallthrough());
+  m.impl("tan_", CppFunction::makeFallthrough());
+  m.impl("tanh", CppFunction::makeFallthrough());
+  m.impl("tanh.out", CppFunction::makeFallthrough());
+  m.impl("tanh_", CppFunction::makeFallthrough());
+  m.impl("threshold", CppFunction::makeFallthrough());
+  m.impl("threshold.out", CppFunction::makeFallthrough());
+  m.impl("threshold_", CppFunction::makeFallthrough());
+  m.impl("to.device", CppFunction::makeFallthrough());
+  m.impl("to.dtype", CppFunction::makeFallthrough());
+  m.impl("to.dtype_layout", CppFunction::makeFallthrough());
+  m.impl("transpose.Dimname", CppFunction::makeFallthrough());
+  m.impl("transpose.int", CppFunction::makeFallthrough());
+  m.impl("true_divide.Scalar", CppFunction::makeFallthrough());
+  m.impl("true_divide.Tensor", CppFunction::makeFallthrough());
+  m.impl("true_divide.out", CppFunction::makeFallthrough());
+  m.impl("true_divide_.Scalar", CppFunction::makeFallthrough());
+  m.impl("true_divide_.Tensor", CppFunction::makeFallthrough());
+  m.impl("trunc", CppFunction::makeFallthrough());
+  m.impl("trunc.out", CppFunction::makeFallthrough());
+  m.impl("trunc_", CppFunction::makeFallthrough());
+  m.impl("unbind.Dimname", CppFunction::makeFallthrough());
+  m.impl("unbind.int", CppFunction::makeFallthrough());
+  m.impl("unflatten.Dimname", CppFunction::makeFallthrough());
+  m.impl("unflatten.int", CppFunction::makeFallthrough());
+  m.impl("uniform_", CppFunction::makeFallthrough());
+  m.impl("vander", CppFunction::makeFallthrough());
+  m.impl("var", CppFunction::makeFallthrough());
+  m.impl("var.dim", CppFunction::makeFallthrough());
+  m.impl("var.names_dim", CppFunction::makeFallthrough());
+  m.impl("var.names_out", CppFunction::makeFallthrough());
+  m.impl("var.out", CppFunction::makeFallthrough());
+  m.impl("var_mean", CppFunction::makeFallthrough());
+  m.impl("var_mean.dim", CppFunction::makeFallthrough());
+  m.impl("var_mean.names_dim", CppFunction::makeFallthrough());
+  m.impl("zero_", CppFunction::makeFallthrough());
+  m.impl("zeros_like", CppFunction::makeFallthrough());
+
+  // These weren't marked as supporting named tensors but were implicitly
+  // supported because they were manually registered.  I'm not sure
+  // if these registrations are right or not, but they preserve old behavior
+  // (and some of them are exercised by the test suite).
+  m.impl("backward", CppFunction::makeFallthrough());
+  m.impl("set_data", CppFunction::makeFallthrough());
+  m.impl("data", CppFunction::makeFallthrough());
+  m.impl("is_leaf", CppFunction::makeFallthrough());
```

Comment:
Does it matter what the second argument (`CppFunction::makeFallthrough()`) is here? A lot of the autograd things have `manual_kernel_registration: True` in native_functions.yaml; from experimentation it looks like no matter whatever function is registered, the catch-all kernel gets called: https://github.com/pytorch/pytorch/blob/c60daedb362a2c93cb5c56a1a0c0851a6388b06d/torch/csrc/autograd/VariableTypeManual.cpp#L398

Reply:
Yes, I believe it matters. The sequence of events that happens on `is_leaf` is that we first invoke at Variable dispatch key (since that is enabled by default and is first in priority), and then when we redispatch with Variable excluded, the next in line would be Named, and NOW we will hit this fallthrough case and keep going to CPU.